@here/harp-geoutils/lib/coordinates/GeoPolygon.ts

```diff
@@ -0,0 +1,205 @@
+            geoBox = GeoBox.fromCoordinates(
+                new GeoCoordinates(south, west),
+                new GeoCoordinates(north, east)
+            );
+        }
+        return geoBox;
+    }
+
+    getCentroid(): GeoCoordinates | undefined {
+        const area = this.getArea();
+        if (area === 0) {
+            return undefined;
+        }
+        let latitude = 0;
+        let longitude = 0;
+        let f;
+
+        let previousIndex = this.coordinates.length - 1;
+
+        this.coordinates.forEach((coordinate, index) => {
+            const previousCoordinate = geoCoordLikeToGeoCoordinatesLike(
+                this.coordinates[previousIndex]
+            );
+            coordinate = geoCoordLikeToGeoCoordinatesLike(coordinate);
+            f =
+                coordinate.latitude * previousCoordinate.longitude -
+                previousCoordinate.latitude * coordinate.longitude;
+            latitude += (coordinate.latitude + previousCoordinate.latitude) * f;
+            longitude += (coordinate.longitude + previousCoordinate.longitude) * f;
+            previousIndex = index;
+        });
+
+        f = area * 6;
+
+        return new GeoCoordinates(latitude / f, area < 0 ? -180 + longitude / f : longitude / f);
+    }
+
+    private sortCCW() {
+        const polyCenter = this.getPolyAverageCenter();
+        if (!polyCenter) {
+            return;
+        }
+
+        //sorts by angle from x-axis
+        this.coordinates.sort((a: GeoCoordLike, b: GeoCoordLike) => {
+            a = geoCoordLikeToGeoCoordinatesLike(a);
+            b = geoCoordLikeToGeoCoordinatesLike(b);
+            const veca = new Vector2(
+                a.latitude - polyCenter.latitude,
+                a.longitude - polyCenter.longitude
+            ).normalize();
+            const vecb = new Vector2(
+                b.latitude - polyCenter.latitude,
+                b.longitude - polyCenter.longitude
+            ).normalize();
+
+            return vecb.angle() - veca.angle();
+        });
+    }
+
+    private getPolyAverageCenter(): GeoCoordinates | undefined {
+        let polySum = this.coordinates.reduce((prev, curr) => {
+            prev = geoCoordLikeToGeoCoordinatesLike(prev);
+            curr = geoCoordLikeToGeoCoordinatesLike(curr);
+            return new GeoCoordinates(
+                prev.latitude + curr.latitude,
+                prev.longitude + curr.longitude
+            );
+        });
+        polySum = geoCoordLikeToGeoCoordinatesLike(polySum);
+        //create an average center point
+        return new GeoCoordinates(
+            polySum.latitude / this.coordinates.length,
+            polySum.longitude / this.coordinates.length
+        );
+    }
+
+    private getArea(): number {
+        let area = 0;
+        let previousIndex = this.coordinates.length - 1;
+
+        this.coordinates.forEach((coordinate, index) => {
+            const previousCoordinate = geoCoordLikeToGeoCoordinatesLike(
```

Comment:
This is creating a lot of temporary object. Please convert the coordinates once in the contructor

Reply:
done

Reply:
done